src/qat/purr/compiler/devices.py

```diff
@@ -5,6 +5,7 @@
 import os
 import sys
 from enum import Enum, auto
+from functools import cached_property
```

Comment:
If we are moving towards `pydantic` functionality in qat, would it be a good idea to start using `QuantumComponent` as a child class of `BaseModel` and use their functionality for cached properties? See [pydantic docu](https://docs.pydantic.dev/2.0/usage/computed_fields/)

Reply:
I'd love to do this but I think we should probably prototype out a switch to pydantic first.